.github/workflows/snapshoty.yml

```diff
@@ -0,0 +1,27 @@
+---
+
+name: snapshoty
+on:
+  workflow_run:
+    workflows:
+      - macos
+    types:
+      - completed
+
+permissions:
+  contents: read
+
+jobs:
+  macos-publish-snapshots:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/download-artifact@v3
+      with:
+        name: snapshoty
```

Comment:
Question, will this step be able to know that the artifacts to be downloaded are the ones generated by the specific commit in the `workflow_run`?

Reply:
Unfortunately, this action isn't able to discover artifacts from another workflow.
The following GitHub Action `https://github.com/dawidd6/action-download-artifact` should do the job.